ssh/tailssh/incubator.go

```diff
@@ -312,6 +290,143 @@ func beIncubator(args []string) error {
+// the shell will use the "login" PAM profile.
+func tryLoginShell(logf logger.Logf, ia incubatorArgs) (bool, error) {
+	euid := os.Geteuid()
+	runningAsRoot := euid == 0
+
+	// Decide whether we should attempt to get a full login shell using either
+	// the login or su commands.
+	attemptLoginShell := true
+	switch {
+	case ia.isSFTP:
+		// If we're going to run an sFTP server, we don't want a shell
+		attemptLoginShell = false
+	case !runningAsRoot:
+		// We have to be root in order to create a login shell.
+		attemptLoginShell = false
+	case hostinfo.IsSELinuxEnforcing():
+		// If we're running on a SELinux-enabled system, neiher login nor su
+		// will be able to set the correct context for the shell. So, we don't
+		// bother trying to run them and instead fall back to using the
+		// incubator to launch the shell.
+		// See http://github.com/tailscale/tailscale/issues/4908.
+		attemptLoginShell = false
+	}
+
+	if !attemptLoginShell {
+		logf("not attempting login shell")
+		return false, nil
+	}
+
+	shouldUseLoginCmd := ia.isShell || runtime.GOOS == "darwin"
+	switch runtime.GOOS {
+	case "linux", "freebsd", "openbsd":
+		if !ia.hasTTY {
+			// We can only use login command if a shell was requested with a TTY. If
+			// there is no TTY, login exits immediately, which breaks things likes
+			// mosh and VSCode.
+			shouldUseLoginCmd = false
+		}
+	}
+
+	if shouldUseLoginCmd {
+		if loginCmdPath, err := exec.LookPath("login"); err == nil {
+			logf("using %s command", loginCmdPath)
+			return true, unix.Exec(loginCmdPath, ia.loginArgs(loginCmdPath), os.Environ())
+		}
+	}
+
+	// We weren't able to use login, maybe we can use su.
+	// Currently, we only support falling back to su on Linux. This
+	// potentially could work on BSDs as well, but requires testing.
+	canUseSU := runtime.GOOS == "linux"
+	if !canUseSU {
+		logf("not attempting su")
+		return false, nil
+	}
+	return tryLoginWithSU(logf, ia)
+}
+
+// tryLoginWithSU attempts to start a login shell using su instead of login. If
+// su is available and supports the necessary arguments, this returns true,
+// plus the result of executing su. Otherwise, it returns false, nil.
+func tryLoginWithSU(logf logger.Logf, ia incubatorArgs) (bool, error) {
+	su, err := exec.LookPath("su")
+	if err != nil {
+		// Can't find su, don't bother trying.
+		logf("can't find su command")
+		return false, nil
+	}
+
+	// Get help text to inspect supported flags.
+	out, err := exec.Command(su, "-h").CombinedOutput()
+	if err != nil {
+		logf("%s doesn't support -h, don't use", su)
+		// Can't even call su -h, don't bother trying.
+		return false, nil
+	}
+
+	supportsFlag := func(flag string) bool {
+		return bytes.Contains(out, []byte(flag))
+	}
+
+	// Make sure su supports the necessary flags.
+	if !supportsFlag("-l") {
+		logf("%s doesn't support -l, don't use", su)
+		return false, nil
+	}
+	if !supportsFlag("-c") {
+		logf("%s doesn't support -c, don't use", su)
+		return false, nil
+	}
```

Comment:
I don't think it is a good idea to search for 2 bytes in the output of `su -h`. This is likely going to break unexpectedly and is not going to be easy to troubleshoot w/o the endusers systems `su -h` output.

Reply:
Alternate suggestions?

Reply:
maybe look for the long form instead like `--login`?